azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecutorServer.java

```diff
@@ -588,4 +580,51 @@ public int getPort() {
   public String getExecutorHostPort() {
     return getHost() + ":" + getPort();
   }
+
+  /**
+   * Shutdown the server.
+   *  - performs a safe shutdown. Waits for completion of current tasks
+   *  - spawns a shutdown thread and returns immediately.
+   */
+  public void shutdown() {
+    logger.warn("Shutting down AzkabanExecutorServer...");
+    new Thread(() -> {
+      try {
+        // Hack: Sleep for a little time to allow API calls to complete
+        Thread.sleep(2000);
+      } catch (InterruptedException e) {
+        logger.error(e);
+      }
+      shutdownInternal();
+    }, "shutdown").start();
+  }
+
+  /**
+   * (internal API)
+   * Note: This should be run in a separate thread.
+   *
+   * Shutdown the server. (blocking call)
+   *  - waits for jobs to finish
+   *  - doesn't accept any new jobs
+   */
+  private void shutdownInternal() {
+    getFlowRunnerManager().shutdown();
+    try {
+      shutdownNow();
```

Comment:
well. shutdown refers to a safe shutdown, while shutdownNow refers to an immediate (may be unsafe) shutdown. This is a default convention for lot of libraries. You can see `java.util.concurrent.ExecutorService#shutdownNow` for example.